### PR TITLE
Fix StatusCode on transaction result

### PR DESCRIPTION
**Pull request by bluesign**
Reported: https://github.com/onflow/flow-go/issues/4494

## Description

______

For contributor use:

- [X] Targeted PR against `master` branch
- [X] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by turbolent** [comment]
@peterargue Could you please have a look and check this matches ANs?

File: emulator/blockchain.go [hunk 1]
```diff
@@ -888,9 +889,13 @@ func (b *Blockchain) GetTransactionResult(txID flowgo.Identifier) (*access.Trans
 		return nil, err
 	}
 
+	statusCode := codes.OK
+	if storedResult.ErrorCode > 0 {
+		statusCode = codes.Canceled
+	}
```
Comment (peterargue) [inline emulator/blockchain.go:895]: The docs that say it's a grpc status code are misleading. I assume that's something to do with the fact that it's accessed over the grpc api (🤷). It's actually one of these enum values
https://github.com/onflow/flow-go/blob/37ece4cd39961442ebae8d622b63fb5a8cf8da18/model/flow/transaction.go#L406-L417

so it should either be `flow.TransactionStatusSealed` or `flow.TransactionStatusExecuted`.

Comment (peterargue) [inline emulator/blockchain.go:895]: The code that sets the status code is here:
https://github.com/onflow/flow-go/blob/37ece4cd39961442ebae8d622b63fb5a8cf8da18/engine/execution/rpc/engine.go#L306

```suggestion
	statusCode := 0
	if storedResult.ErrorCode > 0 {
		statusCode = 1
	}
```

Comment (bluesign) [reply emulator/blockchain.go]: Ah it is not grpc code, thanks

**Review by peterargue — APPROVED** [review]

**Pull request merged**